File: benchmarks/fair/arrayiter.go
package main
import ("fmt";"os";"strconv")
func main(){
    n:=5000000
    if v,err:=strconv.Atoi(os.Getenv("BENCH_N")); err==nil&&v>0 { n=v }
    a:=make([]int64,n); t:=int64(0)
    for i:=0;i<len(a);i++ { a[i]=int64(i) }
    for i:=0;i<len(a);i++ { t+=a[i] }
    fmt.Println(t)
}

File: benchmarks/fair/arrayiter.java
public class arrayiter{
  public static void main(String[] s){
    int n=5000000; String e=System.getenv("BENCH_N");
    if(e!=null){ try{ int v=Integer.parseInt(e); if(v>0)n=v; }catch(Exception x){} }
    long[] a=new long[n]; long t=0;
    for(int i=0;i<a.length;i++) a[i]=i;
    for(int i=0;i<a.length;i++) t+=a[i];
    System.out.println(t);
  }
}
